client/lib/pages/onboarding/country_list_page.dart

```diff
@@ -27,30 +27,49 @@ class CountryListPage extends StatefulWidget {
 
 class _CountryListPageState extends State<CountryListPage> {
   String selectedCountryCode;
+  var listData;
 
   @override
   void initState() {
     super.initState();
-
     selectedCountryCode = widget.selectedCountryCode;
+    listData = (widget.countries?.values ?? []);
   }
 
   @override
   Widget build(BuildContext context) {
     return PageScaffold(
       // TODO: localize?
-      title: 'Country',
+      // title: 'Country',
+      headerWidget: TextField(
+          decoration: InputDecoration(
+              border: InputBorder.none, hintText: 'Enter your country'),
+          onChanged: (String value) async {
+            List filtered = (widget.countries?.values ?? []).where((element) {
+              var listItem = element.name
+                  .toString()
+                  .toUpperCase()
+                  .startsWith(value.toUpperCase());
+              if (listItem || selectedCountryCode == element.alpha2Code) {
+                return true;
+              }
+              return false;
+            }).toList();
+            setState(() {
+              listData = filtered;
+            });
+          }),
       color: Constants.backgroundColor,
-      body: _buildCountries(),
+      body: _buildCountries(listData),
     );
   }
 
-  List<Widget> _buildCountries() {
+  List<Widget> _buildCountries(listData) {
```

Comment:
Revert change because selectedCountries is within scope
```suggestion
  List<Widget> _buildCountries() {
```